src/test/scala/com/fulcrumgenomics/bam/AddReadGroupsByNameTest.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.FgBioDef.PathToBam
+import com.fulcrumgenomics.testing.{SamRecordSetBuilder, UnitSpec}
+import htsjdk.samtools.{SAMFileHeader, SamReaderFactory}
+
+class AddReadGroupsByNameTest extends UnitSpec {
+
+  "RecordInfo" should "throw an exception if a read name is malformed" in {
+    // Fail
+    an[Exception] should be thrownBy RunInfo("")
+    an[Exception] should be thrownBy RunInfo("field")
+    an[Exception] should be thrownBy RunInfo("1:2:3:4:5:6")
+    an[Exception] should be thrownBy RunInfo("1:2:3:4:5:6:7:8")
+    an[Exception] should be thrownBy RunInfo("1:2:3:a:5:6:7")
```

Comment:
Additional tests:
1) More fields after a space as described in the usage doc
2) A "real" read name gathered from a BAM/fastq file